SofaKernel/modules/SofaBaseMechanics/MechanicalObject.inl

```diff
@@ -1524,30 +1535,42 @@ const Data<typename MechanicalObject<DataTypes>::VecCoord>* MechanicalObject<Dat
 {
     if (v.isNull())
     {
-        serr << "Accessing null VecCoord" << sendl;
+        msg_error() << "Reading null VecCoord";
     }
+
     if (v.index < vectorsCoord.size() && vectorsCoord[v.index] != NULL)
     {
         const Data<typename MechanicalObject<DataTypes>::VecCoord>* d = vectorsCoord[v.index];
+        if(d!=NULL)
+        {
 #if defined(SOFA_DEBUG) || !defined(NDEBUG)
-        const typename MechanicalObject<DataTypes>::VecCoord& val = d->getValue();
-        if (!val.empty() && val.size() != (unsigned int)this->getSize())
+            const typename MechanicalObject<DataTypes>::VecCoord& val = d->getValue();
+            if (!val.empty() && val.size() != (unsigned int)this->getSize())
+            {
+                msg_error() << "Accessing State vector " << v << " with incorrect size : " << val.size() << " != " << this->getSize();
+            }
+#endif
+        }
+        else
         {
-            serr << "Accessing State vector " << v << " with incorrect size : " << val.size() << " != " << this->getSize() << sendl;
+            msg_error() << "Reading State vector using a NULL pointer";
         }
-#endif
         return d;
     }
     else
     {
-        serr << "Vector " << v << " does not exist" << sendl;
+        msg_error() << "Vector " << v << " does not exist";
         return NULL;
     }
 }
 
 template <class DataTypes>
 Data<typename MechanicalObject<DataTypes>::VecDeriv>* MechanicalObject<DataTypes>::write(core::VecDerivId v)
 {
+    if (v.isNull())
+    {
+        msg_error() << "Accessing null VecDeriv";
+    }
```

Comment:
What should happen to v if `v.isNull()` ? This block does not pass in scene tests.

Reply:
I tried to understand but...
So when v.isNull, this means that v.index == 0.

So in the next lines, it enters this block for the first time and create a vector.
```
if (vectorsDeriv[v.index] == NULL)
{
        vectorsDeriv[v.index] = new Data< VecDeriv >;
        vectorsDeriv[v.index]->setName(v.getName());
        vectorsDeriv[v.index]->setGroup("Vector");
```
For example In the scene: examples/Components/collision/RuleBasedContactManager.scn this error happened when constraints are created here:
```
template<class DataTypes>
void PairInteractionConstraint<DataTypes>::storeLambda(const ConstraintParams* cParams, MultiVecDerivId res, const sofa::defaulttype::BaseVector* lambda)
{
    if (cParams)
    {
        storeLambda(cParams, *res[mstate1.get(cParams)].write(), *res[mstate2.get(cParams)].write(), *cParams->readJ(mstate1), *cParams->readJ(mstate2), lambda);
    }
}
```

but the scene seems to act normally... (it moves)

Reply:
I find it logical that a new Vecid is created when contact happened for the first time, but I don't understand why index is null.